sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/BeamSqlEnv.java

```diff
@@ -73,83 +70,18 @@ public static BeamSqlEnv inMemory(TableProvider... tableProviders) {
-      for (Method method : entry.getValue()) {
-        connection.getCurrentSchemaPlus().add(entry.getKey(), UdfImpl.create(method));
-      }
-    }
-  }
-
-  public void addSchema(String name, TableProvider tableProvider) {
-    connection.setSchema(name, tableProvider);
-  }
-
-  public void setCurrentSchema(String name) {
-    try {
-      connection.setSchema(name);
-    } catch (SQLException e) {
-      throw new RuntimeException(e);
-    }
-  }
-
-  /** Register a UDF function which can be used in SQL expression. */
-  public void registerUdf(String functionName, Class<?> clazz, String method) {
-    connection.getCurrentSchemaPlus().add(functionName, UdfImpl.create(clazz, method));
-  }
-
-  /** Register a UDF function which can be used in SQL expression. */
-  public void registerUdf(String functionName, Class<? extends BeamSqlUdf> clazz) {
-    registerUdf(functionName, clazz, BeamSqlUdf.UDF_METHOD);
-  }
-
-  /**
-   * Register {@link SerializableFunction} as a UDF function which can be used in SQL expression.
-   * Note, {@link SerializableFunction} must have a constructor without arguments.
-   */
-  public void registerUdf(String functionName, SerializableFunction sfn) {
-    registerUdf(functionName, sfn.getClass(), "apply");
-  }
-
-  /**
-   * Register a UDAF function which can be used in GROUP-BY expression. See {@link
-   * org.apache.beam.sdk.transforms.Combine.CombineFn} on how to implement a UDAF.
-   */
-  public void registerUdaf(String functionName, Combine.CombineFn combineFn) {
-    connection.getCurrentSchemaPlus().add(functionName, new UdafImpl(combineFn));
-  }
-
-  /** Load all UDF/UDAF from {@link UdfUdafProvider}. */
-  public void loadUdfUdafFromProvider() {
-    ServiceLoader.<UdfUdafProvider>load(UdfUdafProvider.class)
-        .forEach(
-            ins -> {
-              ins.getBeamSqlUdfs().forEach((udfName, udfClass) -> registerUdf(udfName, udfClass));
-              ins.getSerializableFunctionUdfs()
-                  .forEach((udfName, udfFn) -> registerUdf(udfName, udfFn));
-              ins.getUdafs().forEach((udafName, udafFn) -> registerUdaf(udafName, udafFn));
-            });
-  }
-
-  public void loadBeamBuiltinFunctions() {
-    for (BeamBuiltinFunctionProvider provider :
-        ServiceLoader.load(BeamBuiltinFunctionProvider.class)) {
-      registerBuiltinUdf(provider.getBuiltinMethods());
-    }
-  }
-
   public BeamRelNode parseQuery(String query) throws ParseException {
     try {
       return planner.convertToBeamRel(query);
-    } catch (ValidationException | RelConversionException | SqlParseException e) {
+    } catch (Exception e) {
       throw new ParseException(String.format("Unable to parse query %s", query), e);
     }
   }
 
   public boolean isDdl(String sqlStatement) throws ParseException {
     try {
       return planner.parse(sqlStatement) instanceof SqlExecutableStatement;
-    } catch (SqlParseException e) {
+    } catch (Exception e) {
```

Comment:
This way we're also wrapping runtime exceptions, is there a reason for this?

Reply:
Right now it's less clear what exception should be throw from `QueryPlanner` interface, so I use a general `Exception` here. 

I can change the exception back to what originally here it is by changing `QueryPlanner` interface. By doing so `QueryPlanner` interface can only be used for `CalciteQueryPlanner` because it throws Calcite's exceptions. I should add experimental annotation on it.

Later when adding any new planners, we can gradually generalize the exception, probably starting from defining BeamSQL exceptions and let the interface throws that.

Is there better option?